webrtc.html

```diff
@@ -2582,6 +2636,110 @@
       </dl>
     </section>
 
+    <section>
+      <h2 id="sec.cert-mgmt">Certificate Management</h2>
+      <p>
+        The certificates that <code>RTCPeerConnection</code> instances use to
+        authenticate with peers use the <a><code>RTCCertificate</code></a>
+        interface.  These objects can be explicitly generated by applications
+        using <code>RTCPeerConnection.generateCertificate</code> and provided in
+        the <a><code>RTCConfiguration</code></a> when constructing a
+        new <code>RTCPeerConnection</code> instance.
+      </p>
+      <p>
+        The explicit certificate management functions provided here are
+        optional.  If an application does not provide
+        the <code>certificates</code> configuration option when constructing
+        an <code>RTCPeerConnection</code> a new set of certificates MUST be
+        generated by the <a>user agent</a>.  That set MUST include an ECDSA
+        certificate with a private key on the P-256 curve and a signature with a
+        SHA-256 hash.
+      </p>
+
+      <dl class="idl" title="partial interface RTCPeerConnection">
+        <dt>static Promise&lt;RTCCertificate&gt; generateCertificate
+          (AlgorithmIdentifier keygenAlgorithm)</dt>
+        <dd>
+          <p>This function takes an argument that describes how the private key associated with the certificate is generated.  The <var>keygenAlgorithm</var> argument uses the WebCrypto
+          [[!WebCryptoAPI]] <a href="https://dvcs.w3.org/hg/webcrypto-api/raw-file/tip/spec/Overview.html#dfn-AlgorithmIdentifier">AlgorithmIdentifier</a>
+          type.  The <var>keygenAlgorithm</var> value MUST be a valid argument
+            to <a href="https://dvcs.w3.org/hg/webcrypto-api/raw-file/tip/spec/Overview.html#SubtleCrypto-method-generateKey"><code>Crypto.subtle.generateKey</code></a>.</p>
```

Comment:
I'd be surprised if the TAG doesn't flag it, so it may be worth pre-emptively engaging them (and WebCrypto) for how we can facilitate this.

This sentence establishes a normative requirement, but doesn't actually provide a description of what that normative requirement means. For example, an implementation COULD interpret "a valid argument" to mean "conforms to the AlgorithmIdentifier" WebIDL specification.

At a minimum, you should reference the algorithm from WebCrypto for normalizing an algorithm ( https://dvcs.w3.org/hg/webcrypto-api/raw-file/tip/spec/Overview.html#algorithm-normalization-normalize-an-algorithm ), wherein the input is an AlgorithmIdentifier alg and an operation "op"

A rough stab at how that might be expressed algorithmically

1. Let *foo* be the result of performing the (normalize an algorithm)[https://dvcs.w3.org/hg/webcrypto-api/raw-file/tip/spec/Overview.html#algorithm-normalization-normalize-an-algorithm] algorithm, with `alg` set to *keygenAlgorithm* and `op` set to **generateKey**.
2. Let *bar* be the result of performing the (normalize an algorithm)[https://dvcs.w3.org/hg/webcrypto-api/raw-file/tip/spec/Overview.html#algorithm-normalization-normalize-an-algorithm] algorithm, with `alg` set to *keygenAlgorithm* and `op` set to **sign**.
3. (something something about rejecting the promise if an error is thrown, or maybe that's maybe that's implicit in Promise IDL handling because I can never remember)
4. (something something about performing a deep equality check)

Of course, the proposed API **does not work** for valid use cases. For example, for RSASSA-PKCS1-v1_5, the input is a https://dvcs.w3.org/hg/webcrypto-api/raw-file/tip/spec/Overview.html#dfn-RsaHashedKeyGenParams (a dictionary with keys for *name*, *modulusLength*, *publicExponent*, *hash*), but the sign parameters are null. For RSA-PSS, you have https://dvcs.w3.org/hg/webcrypto-api/raw-file/tip/spec/Overview.html#dfn-RsaHashedKeyGenParams vs https://dvcs.w3.org/hg/webcrypto-api/raw-file/tip/spec/Overview.html#dfn-RsaPssParams

If you meant for the AlgorithmIdentifier to just be a proxy for "name", then you probably simply want to be looking up the `name` key of *keygenAlgorithm*, and then dealing with the **[supportedAlgorithms](https://dvcs.w3.org/hg/webcrypto-api/raw-file/tip/spec/Overview.html#dfn-supportedAlgorithms)** internal slot in some way.

To put it differently, I have *no* idea how such a combination of specs would be implemented in Chrome, nor am I sure how best to recommend text that could be implemented.